api/src/test/java/org/openmrs/module/fhir2/api/search/DiagnosticReportSearchQueryImplTest.java

```diff
@@ -333,6 +335,24 @@ public void searchForDiagnosticReports_shouldReturnCorrectObsByPatientName() {
 		    equalTo(DIAGNOSTIC_REPORT_UUID));
 	}
 	
+	@Test
+	public void searchForDiagnosticReports_shouldSearchForUniqueObsByPatientName() {
+		ReferenceParam patientReference = new ReferenceParam(Patient.SP_NAME, "Horatio Hornblower");
+		ReferenceAndListParam patientList = new ReferenceAndListParam();
+		patientList.addValue(new ReferenceOrListParam().add(patientReference));
+		
+		SearchParameterMap theParams = new SearchParameterMap().addParameter(FhirConstants.PATIENT_REFERENCE_SEARCH_HANDLER,
+		    patientList);
+		
+		IBundleProvider results = search(theParams);
+		
+		assertThat(results, notNullValue());
+		assertThat(results.size(), equalTo(1));
+		
+		Set<String> resultSet = new HashSet<>(dao.getResultUuids(theParams));
+		assertThat(resultSet.size(), equalTo(1)); // 3 with repetitions
```

Comment:
Remove the comments

Reply:
I think the comments here justify that we are indeed not getting repeated results and should be kept. What do you think?

Reply:
That's fine with me 👌